iina/VideoView.swift

```diff
@@ -336,9 +331,16 @@ class VideoView: NSView {
       }
     }
 
-    if videoLayer.colorspace != VideoView.SRGB {
-      videoLayer.colorspace = VideoView.SRGB
-      videoLayer.wantsExtendedDynamicRangeContent = false
+    let screenColorSpace = player.mainWindow.window?.screen?.colorSpace
```

Comment:
I'm operating under the assumption that the picture displayed by IINA must match the picture displayed by mpv (with matching options and mpv not using the new code that is not available to IINA) and therefore IINA should treat mpv as the reference implementation and thus IINA's code should as much as possible match up with how mpv uses `libmpv`.

I was very undecided on some of these differences and finally decided it was better to align with mpv and if there are problems bring them to the attention of mpv. I'm still unsure about that decision, so good to question all this. I am worried about how much to match the mpv code as we already decided the `init(layer:)` code was incorrect.

One difference I did not change is that IINA enables the multi-threaded OpenGL engine by setting `kCGLCEMPEngine` in the OpenGL context.

On the color space of the layer… 

I have added code to set the color space of the layer to the color space of the display since that is what mpv does as seen [here](https://github.com/mpv-player/mpv/blob/02254b92dd237f03aa0a151c2a68778c4ea848f9/video/out/cocoa_cb_common.swift#L118-L126).

If I comment out setting the color space then it ends up with a value of `kCGColorSpaceExtendedLinearSRGB` and the picture looks like this:
![colorspace-not-set](https://github.com/user-attachments/assets/5f13d4ed-0081-4c33-8b77-1495733a0030)

Reply:
I had commented that line out as well. I confirmed it was `nil` at the end of the initializer. I failed trying to figure out how to set a watch point on Xcode to see where that is getting set. Maybe I missed something. I will debug some more to confirm I did not mess up that test.

Reply:
From a very quick test, that statement sets the color space to `kCGColorSpaceDeviceRGB`. Something else is setting it to `kCGColorSpaceExtendedLinearSRGB`.

Reply:
I fought with Xcode and lldb quite a bit and struck out. I was able to set a breakpoint on `setColorSpace` and it was not hit. It seemed like the problem was that Xcode did not find such a method on `CAOpenGLLayer`. I tried to step into an assignment to `colorspace` and that caused the debugger to fail. So I still have not found what is setting it to `kCGColorSpaceExtendedLinearSRGB`.

Reply:
I didn't get an error. I was trying to set a breakpoint on all `setColorSpace` methods. It worked but for some reason did not set one on the `CAOpenGLLayer` method. The problem with stepping into was strange. The Xcode `Debug` menu looked like it does when the app is running, with most items greyed out. Hovering over IINA showed the spinning beach ball of death.

I was able to set the breakpoint and the color space is being set when drawing if I comment out the code in IINA that set it:
```text
(lldb) bt
* thread #1, stop reason = breakpoint 2.1
  * frame #0: 0x00000001b1e1dd98 QuartzCore`-[CAOpenGLLayer setColorspace:]
    frame #1: 0x00000001b1c35710 QuartzCore`CAOpenGLLayerDraw(CAOpenGLLayer*, double, CVTimeStamp const*, unsigned int) + 540
    frame #2: 0x00000001b1c35324 QuartzCore`-[CAOpenGLLayer _display] + 520
    frame #3: 0x000000010538f034 IINA`ViewLayer.display(self=0x000060000273dd50) at ViewLayer.swift:209:11
    frame #4: 0x0000000105390130 IINA`@objc ViewLayer.display() at <compiler-generated>:0
    frame #5: 0x00000001b1ba7ae8 QuartzCore`CA::Layer::display_if_needed(CA::Transaction*) + 760
    frame #6: 0x00000001b1d1b4d4 QuartzCore`CA::Context::commit_transaction(CA::Transaction*, double, double*) + 456
    frame #7: 0x00000001b1b8a2c0 QuartzCore`CA::Transaction::commit() + 648
    frame #8: 0x000000010538f054 IINA`ViewLayer.display(self=0x000060000273dd50) at ViewLayer.swift:210:19
    frame #9: 0x0000000105390130 IINA`@objc ViewLayer.display() at <compiler-generated>:0
    frame #10: 0x0000000105390310 IINA`ViewLayer.draw(forced=true, self=0x000060000273dd50) at ViewLayer.swift:256:5
    frame #11: 0x0000000105473400 IINA`PlayerCore.playbackRestarted(self=0x000000013e77ef90) at PlayerCore.swift:1884:37
    frame #12: 0x00000001054fbba4 IINA`closure #8 in MPVController.handleEvent(self=0x0000600002226c00) at MPVController.swift:1101:16
    frame #13: 0x0000000104ff3c94 IINA`thunk for @escaping @callee_guaranteed () -> () at <compiler-generated>:0
    frame #14: 0x0000000109d44e30 libdispatch.dylib`_dispatch_call_block_and_release + 32
    frame #15: 0x0000000109d4699c libdispatch.dylib`_dispatch_client_callout + 20
    frame #16: 0x0000000109d59e24 libdispatch.dylib`_dispatch_main_queue_drain + 1220
    frame #17: 0x0000000109d59950 libdispatch.dylib`_dispatch_main_queue_callback_4CF + 44
    frame #18: 0x00000001aa6dfb84 CoreFoundation`__CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 16
    frame #19: 0x00000001aa69d304 CoreFoundation`__CFRunLoopRun + 1992
    frame #20: 0x00000001aa69c3e8 CoreFoundation`CFRunLoopRunSpecific + 612
    frame #21: 0x00000001b3ef2df0 HIToolbox`RunCurrentEventLoopInMode + 292
    frame #22: 0x00000001b3ef2c2c HIToolbox`ReceiveNextEventCommon + 648
    frame #23: 0x00000001b3ef2984 HIToolbox`_BlockUntilNextEventMatchingListInModeWithFilter + 76
    frame #24: 0x00000001ad8c4134 AppKit`_DPSNextEvent + 636
    frame #25: 0x00000001ad8c32d0 AppKit`-[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 716
    frame #26: 0x00000001ad8b7734 AppKit`-[NSApplication run] + 464
    frame #27: 0x00000001ad88eb84 AppKit`NSApplicationMain + 880
    frame #28: 0x0000000105218d18 IINA`main at AppDelegate.swift:24:7
    frame #29: 0x00000001aa267fd8 dyld`start + 2412
```